lib/internal/fs/utils.js

```diff
@@ -293,7 +293,13 @@ function validateOffsetLengthRead(offset, length, bufferLength) {
   let err;
 
   if (offset < 0 || offset >= bufferLength) {
-    err = new ERR_OUT_OF_RANGE('offset', `>= 0 && <= ${bufferLength}`, offset);
+    if (bufferLength === 0) {
+      throw ERR_INVALID_ARG_VALUE('buffer', 0,
```

Comment:
Please change this to an `ERR_OUT_OF_RANGE` error that corresponds to the `bufferLength`. I do not know what `buffer` should stand for in this case and invalid argument value errors are normally used in different contexts. Here it should definitely be an out of range error as I also suggested in my code example in my earlier comment.